### PR TITLE
[TOOLS-2263] Allowing inputting Fix Version when transition requires

### PR DESCRIPTION
**Pull request by jytakano**

### CODE REVIEW

File: core/src/main/java/objective/taskboard/data/IssueScratch.java [hunk 1]
```diff
@@ -40,7 +42,8 @@
     protected TaskboardTimeTracking timeTracking;
     protected String releaseId;
     protected List<Changelog> changelog; //NOSONAR
-    
+    protected List<Version> fixVersions;
```
Comment (objective-solutions-user) [inline core/src/main/java/objective/taskboard/data/IssueScratch.java]: CRITICAL Make "fixVersions" private or transient. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS1948)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 1 issue
* ![CRITICAL][CRITICAL] 1 critical

Watch the comments in this conversation to review them.

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'

File: .gitignore [hunk 2]
```diff
@@ -3,7 +3,7 @@ dist
 tmp
 out-tsc
 target
-/data
```
[the remark this answers is not in the document]
Comment (jytakano) [reply .gitignore]: Ok

File: core/src/main/java/objective/taskboard/jira/JiraService.java [hunk 3]
```diff
@@ -149,6 +150,12 @@ public String getResolutions(String transitionName) {
         return null;
     }
 
+    public List<FieldsRequiredInTransition> getFieldsRequiredInTransitions(String issueKey, List<Long> transitionIds) {
+        log.debug("⬣⬣⬣⬣⬣  getFieldsRequiredInTransitions");
+        Iterable<FieldsRequiredInTransition> response = jiraEndpointAsUser.request(FieldsRequiredInTransition.Service.class).post(issueKey, transitionIds);
+        return ImmutableList.copyOf(response);
```
[the remark this answers is not in the document]
Comment (jytakano) [reply core/src/main/java/objective/taskboard/jira/JiraService.java]: Ok

File: jira-client/src/main/java/objective/taskboard/jira/data/Transition.java [hunk 4]
```diff
@@ -22,10 +25,34 @@ public Transition(Long id, String name, Status to, Map<String, Field> fields) {
...
     public static class Field {
         public String name;
+        public FieldSchema schema;
         public Boolean required;
         public List<Object> allowedValues;
         public String defaultValue;
         public Boolean hasDefaultValue;
+
+        @JsonIgnore
+        public Boolean isArrayOfVersion() {
```
[the remark this answers is not in the document]
Comment (jytakano) [reply jira-client/src/main/java/objective/taskboard/jira/data/Transition.java]: Ok

File: jira-client/src/main/java/objective/taskboard/jira/data/Transition.java [hunk 5]
```diff
@@ -10,7 +12,8 @@
     public Status to;
     public Map<String, Field> fields;
```
[the remark this answers is not in the document]
Comment (jytakano) [reply jira-client/src/main/java/objective/taskboard/jira/data/Transition.java]: Ok

File: jira-client/src/main/java/objective/taskboard/jira/data/Transition.java [hunk 6]
```diff
@@ -10,7 +12,8 @@
...
     public Long order;
-    
+    public String errorMessage;
```
[the remark this answers is not in the document]
Comment (jytakano) [reply jira-client/src/main/java/objective/taskboard/jira/data/Transition.java]: Ok

**Pull request merged**